eif_utils/src/lib.rs

```diff
@@ -60,25 +61,73 @@ impl SignEnclaveInfo {
     }
 }
 
+/// Utility function to calculate PCRs, used at build and describe.
+pub fn get_pcrs<T: Digest + Debug + Write + Clone>(
+    image_hasher: &mut EifHasher<T>,
+    bootstrap_hasher: &mut EifHasher<T>,
+    app_hasher: &mut EifHasher<T>,
+    cert_hasher: &mut EifHasher<T>,
+    hasher: T,
+    is_signed: bool,
+) -> Result<BTreeMap<String, String>, String> {
+    let mut measurements = BTreeMap::new();
+    let image_hasher = hex::encode(
+        image_hasher
+            .tpm_extend_finalize_reset()
+            .map_err(|e| format!("Could not get result for image_hasher: {:?}", e))?,
+    );
+    let bootstrap_hasher = hex::encode(
+        bootstrap_hasher
+            .tpm_extend_finalize_reset()
+            .map_err(|e| format!("Could not get result for bootstrap_hasher: {:?}", e))?,
+    );
+    let app_hash = hex::encode(
+        app_hasher
+            .tpm_extend_finalize_reset()
+            .map_err(|e| format!("Could not get result for app_hasher: {:?}", e))?,
+    );
+
+    // Hash certificate only if signing key is set, otherwise related PCR will be zero
+    let cert_hash = if is_signed {
+        Some(hex::encode(
+            cert_hasher
+                .tpm_extend_finalize_reset()
+                .map_err(|e| format!("Could not get result for cert_hash: {:?}", e))?,
+        ))
+    } else {
+        None
+    };
+
+    measurements.insert("HashAlgorithm".to_string(), format!("{:?}", hasher));
+    measurements.insert("PCR0".to_string(), image_hasher);
+    measurements.insert("PCR1".to_string(), bootstrap_hasher);
+    measurements.insert("PCR2".to_string(), app_hash);
+    if let Some(cert_hash) = cert_hash {
+        measurements.insert("PCR8".to_string(), cert_hash);
+    }
+
+    Ok(measurements)
+}
+
 pub struct EifBuilder<T: Digest + Debug + Write + Clone> {
     kernel: File,
     cmdline: Vec<u8>,
     ramdisks: Vec<File>,
-    sign_info: Option<SignEnclaveInfo>,
+    pub sign_info: Option<SignEnclaveInfo>,
```

Comment:
Let's avoid doing this, I know it is simpler, but we should add getters if we need this information.